README.md

```diff
@@ -64,3 +64,38 @@ You don't always need this, review which features need the polyfill here: [ES6 F
 var esTranspiler = require('broccoli-babel-transpiler');
 var scriptTree = esTranspiler(inputTree, { browserPolyfill: true });
 ```
+
+## Plugins
+
+Use of custom plugins works similarly to `babel` itself. You would pass a `plugins` array in `options`:
+
+```js
+var esTranspiler = require('broccoli-babel-transpiler');
+var applyFeatureFlags = require('babel-plugin-feature-flags');
+
+var featureFlagPlugin = applyFeatureFlags({
+  import: { module: 'ember-metal/features' },
+  features: {
+    'ember-metal-blah': true
+  }
+});
+
+var scriptTree = esTranspiler(inputTree, {
+  plugins: [
+    featureFlagPlugin
+  ]
+});
+```
+
+### Caching
+
+broccoli-babel-transpiler uses a persistent cache to enable rebuilds to be significantly faster (by avoiding transpilation for files that have not changed).
+However, since a plugin can do many things to affect the transpiled output it must also influence the cache key to ensure transpiled files are rebuilt
+if the plugin changes (or the plugins configuration).
+
+In order to aid plugin developers in this process, broccoli-babel-transpiler will invoke two methods on a plugin so that it can augment the cache key:
+
+* `cacheKey` - This method can return any string value to be used in the cache key calculation. This is generally only needed when the configuration provided
```

Comment:
rather then  describing the property with itself

> cacheKey only matters if the plugin affects the cache key

We may want to say:
```
cacheKey is used to describe any runtime information that may want to invalidate the cache. An example of this would be:

* FEATURE flags for defeatureify
* template-compiler.js for htmlbars pre compiler
* options


```